### PR TITLE
Access to the app with required authentication

### PR DESCRIPTION
**Pull request by benoitblanc**
Hello @pka 

I need to access to the application only when user is logged in. I used what you have done in https://github.com/qwc-services/qwc-admin-gui/commit/fb5955d708a943c74e0373f606d3e13672604dc7 to test if user is logged. Otherwise, user is redirected to authentication page.

Also, when no `TENANT_HEADER` is set in configuration, there is an error that occurs because `//` does'nt work in url. I replace it with a simple `/`. It is the same in [qwc-admin-gui](https://github.com/qwc-services/qwc-admin-gui/).

Thanks

### CODE REVIEW

**Comment by benoitblanc** [comment]
Hello @manisandro ,

I saw that you have fixed the `auth_path_prefix` computation in https://github.com/qwc-services/qwc-admin-gui/commit/4cbaf9542b9808063ae83248e909a27f12ae5845 so I did the same way here.

Thanks

**Comment by manisandro** [comment]
Thanks @benoitblanc , can you rebase your commits, I have since updated the requirements.

**Comment by benoitblanc** [comment]
@manisandro I am not very comfortable with rebase but I hope it will be good now :-)

File: server.py [hunk 1]
```diff
@@ -2,13 +2,15 @@
...
+from flask import json, Flask, request, jsonify, redirect
 from flask_jwt_extended import jwt_optional, get_jwt_identity
 
 from qwc_services_core.jwt import jwt_manager
-from qwc_services_core.tenant_handler import TenantHandler
+from qwc_services_core.tenant_handler import TenantHandler, TenantPrefixMiddleware, TenantSessionInterface
 from qwc2_viewer import QWC2Viewer
 
+AUTH_PATH = os.environ.get('AUTH_PATH', '/auth')
+AUTH_REQUIRED = os.environ.get('AUTH_REQUIRED', False)
```
Comment (manisandro) [inline server.py:13]: We used

     os.environ.get('xxx', '0') not in [0, "0", "False", "FALSE"]

to handle bool env-vars elsewhere.

File: requirements.txt [hunk 2]
```diff
@@ -3,4 +3,4 @@ Flask-JWT-Extended==3.25.0
 flask-restx==0.2.0
 requests==2.23.0
 werkzeug==0.16.1
-qwc-services-core==1.2.4
+qwc-services-core==1.2.4
```
Comment (manisandro) [inline requirements.txt:6]: This is an no-change change, can you remove it from the PR?

**Comment by manisandro** [comment]
Hi @benoitblanc , I've added two comments, rest looks good.

**Comment by benoitblanc** [comment]
Hi @manisandro , I think I have resolved all your remarks. Thanks !

**Comment by manisandro** [comment]
Thanks!

**Pull request merged**